x-pack/legacy/plugins/actions/server/create_execute_function.ts

```diff
@@ -6,34 +6,56 @@
 
 import { SavedObjectsClientContract } from 'src/core/server';
 import { TaskManager } from '../../task_manager';
-import { SpacesPlugin } from '../../spaces';
+import { GetBasePathFunction } from './types';
 
 interface CreateExecuteFunctionOptions {
   taskManager: TaskManager;
-  internalSavedObjectsRepository: SavedObjectsClientContract;
-  spaceIdToNamespace: SpacesPlugin['spaceIdToNamespace'];
+  getScopedSavedObjectsClient: (request: any) => SavedObjectsClientContract;
+  getBasePath: GetBasePathFunction;
 }
 
 export interface ExecuteOptions {
   id: string;
   params: Record<string, any>;
   spaceId: string;
+  apiKeyId: string | null;
+  apiKeyValue: string | null;
 }
 
 export function createExecuteFunction({
+  getBasePath,
+  getScopedSavedObjectsClient,
   taskManager,
-  internalSavedObjectsRepository,
-  spaceIdToNamespace,
 }: CreateExecuteFunctionOptions) {
-  return async function execute({ id, params, spaceId }: ExecuteOptions) {
-    const namespace = spaceIdToNamespace(spaceId);
-    const actionSavedObject = await internalSavedObjectsRepository.get('action', id, { namespace });
+  return async function execute({ id, params, spaceId, apiKeyId, apiKeyValue }: ExecuteOptions) {
+    const requestHeaders: Record<string, string> = {};
+
+    if (apiKeyId && apiKeyValue) {
```

Comment:
There's no scenario from reading the docs that would cause one or the other to be null. But thinking of it, we could provide this file a `isApiKeysRequired: boolean` and throw if ever one of the two attributes is missing. We could do similar in other files that do this check as well. This could possibly also catch scenarios where security is dynamically enabled and some alerts are missing API keys.

Reply:
I ended up using `useApiKey` for that scenario instead of the "required" terminology. More than happy to change it to `isSecurityEnabled` instead.

Reply:
Yeah I think isSecurityEnabled makes it clearer _why_ we're using the API key here.